### PR TITLE
Bug 1978744: Add docs explaining automatedCleaningMode overrides

### PR DESCRIPTION
**Pull request by mkowalski**
This is a backport of #2146

### CODE REVIEW

File: docs/hive-integration/baremetal-agent-controller.md [hunk 1]
```diff
@@ -198,6 +198,11 @@ It is possible to specify `RootDeviceHints` for the `BareMetalHost` resource. Ro
 used to tell the installer what disk to use as the installation disk. Refer to the
 [baremetal-operator documentation](https://github.com/metal3-io/baremetal-operator/blob/master/docs/api.md#rootdevicehints) to know more.
 
+---
+**NOTE**
+
+We are always setting `automatedCleaningMode: disabled` even if the `BareMetalHost` manifest specifies another value (e.g. `automatedCleaningMode: metadata`). This may be changed in the future releases, but currently we do not support using Ironic to clean the node.
```
Comment (rollandf) [inline docs/hive-integration/baremetal-agent-controller.md:204]: Who is `we`?
Maybe "BMAC will always set ...."

Comment (mkowalski) [reply docs/hive-integration/baremetal-agent-controller.md:204]: Agree, but this is already a backport PR... How should we handle it? If we change it only here and right now, the git commit history will diverge and we will/may/might have problems in the future

Comment (flaper87) [reply docs/hive-integration/baremetal-agent-controller.md:204]: > Who is we?

Good catch, @rollandf  I completely missed it in the original PR. Let's get this in as-is for now

**Review by flaper87 — APPROVED** [review]

File: docs/hive-integration/baremetal-agent-controller.md (hunk 1 above)
Comment (rollandf) [reply docs/hive-integration/baremetal-agent-controller.md:204]: Ack

Comment (mkowalski) [reply docs/hive-integration/baremetal-agent-controller.md:204]: https://github.com/openshift/assisted-service/pull/2162

**Pull request merged**